File: Problem_7/Problem7.java
import java.util.ArrayList;


public class Problem7 {

    public static void main(String[] args) {
        ArrayList<Integer> primes = new ArrayList<Integer>();
        int i = 3;
        boolean isPrime = true;
        primes.add(2);

        while (primes.size() < 10001) {
            for (Integer prime: primes) {
                isPrime = true;
                if (i%prime == 0) {
                    i += 2;
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                primes.add(i);
            }
        }
        System.out.println(primes.get(primes.size()-1));
    }
}

File: Problem_7/problem7.js
const primes = [2]
let i = 3

while (primes.length < 10001) {
    let isPrime = primes.every((prime) => {
        if (i%prime == 0) {
            i += 2
            return false
        }
        return true
    })
    if (isPrime) primes.push(i)

}

console.log(primes[primes.length -1])